audio-recorder.js

```diff
@@ -86,25 +87,15 @@ module.exports = class AudioRecorder {
     // Detect audio recording capabilities.
     // http://caniuse.com/#feat=stream
     // https://developer.mozilla.org/en-US/docs/Web/API/Navigator.getUserMedia
-    navigator.getUserMedia = navigator.getUserMedia ||
-      navigator.webkitGetUserMedia ||
-      navigator.mozGetUserMedia ||
-      navigator.msGetUserMedia;
-    this.canGetUserMedia = Boolean(navigator.getUserMedia);
-    console.log('Native deprecated navigator.getUserMedia API capability: ' +
+    getUserMedia({audio: true, video: true})
+      .then(() => {
+        this.canGetUserMedia = true;
+      }).catch(() => {
+        this.canGetUserMedia = false;
+      });
+    console.log('Native getUserMedia API capability: ' +
```

Comment:
You made this check asynchronous, so it should be treated as such.

`this.canGetUserMedia` is always undefined.

Put the rest of this function in the promise chain and return it.